src/Illuminate/Mail/Mailer.php

```diff
@@ -392,7 +390,11 @@ protected function sendSwiftMessage($message)
         }
 
         if (! $this->pretending) {
-            return $this->swift->send($message, $this->failedRecipients);
+            try {
+                return $this->swift->send($message, $this->failedRecipients);
+            } finally {
+                $this->getSwiftMailer()->getTransport()->stop();
```

Comment:
Kind of inconsistent here anyway since Mailer::forceReconnection() uses the public getter, but I've changed it according to your comment.